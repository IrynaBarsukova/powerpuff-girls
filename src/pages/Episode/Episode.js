import React, { useEffect } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from "react-redux";

import Container from '../../components/common/Container/Container';
import EntityDetails from '../../components/EntityDetails/EntityDetails';
import { loadEpisodeStarted, clearEpisode } from '../../store/actions/episode';
import styles from  './Episode.scss';

function Episode() {
    const { episodeId } = useParams();
    const history = useHistory();
    const dispatch = useDispatch();
    const episode = useSelector(state => state.episode.data);
    const isLoaded = useSelector(state => state.episode.loaded);


    useEffect(() => {
        dispatch(loadEpisodeStarted(episodeId));

        return () => dispatch(clearEpisode());
    }, [episodeId]);

    const goBack = () => {
        history.goBack();
    };

    return (
        <div className={styles.episode}>
            {isLoaded &&
                <Container>
                    <button className={styles.episode__button} type="button" onClick={goBack}>
                        Go back
                    </button>
                    <EntityDetails entity={episode} />
                </Container>
            }
        </div>
    );
}

export default Episode;
