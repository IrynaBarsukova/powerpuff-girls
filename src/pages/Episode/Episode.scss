@import '../../styles/variables';

.episode {
  margin-top: 20px;
}

.episode__button {
  width: 60px;
  color: $pink;
  font-size: 14px;
  text-align: left;
  background: $white;
  padding: 5px 0;
  border: none;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
}
