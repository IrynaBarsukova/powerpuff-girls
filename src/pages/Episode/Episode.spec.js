import React from 'react';
import { Provider } from 'react-redux';
import { fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import MockStore from '../../mocks/store.mock';
import renderWithRouter from '../../mocks/router.mock';
import mockEpisode from '../../mocks/episode.mock';
import Episode from './Episode';

describe('<Episode />', () => {
    let store;

    beforeEach(() => {
        store = new MockStore({ episode: { data: mockEpisode, loaded: true } });
    });

    it('should render Episode', () => {
        const { container } = renderWithRouter(
            <Provider store={store}>
                <Episode />
            </Provider>,
            { route: `shows/6771/episodes/${mockEpisode.id}` },
        );

        expect(container).toMatchSnapshot();
    });

    it('should handle Go Back button click', () => {
        const mockHistory = {
            ...createMemoryHistory({
                initialEntries: [`shows/6771/episodes/${mockEpisode.id}`],
            }),
            goBack: jest.fn(),
        };
        const { getByText } = renderWithRouter(
            <Provider store={store}>
                <Episode />
            </Provider>,
            { history: mockHistory },
        );

        fireEvent.click(getByText('Go back'));
        expect(mockHistory.goBack).toHaveBeenCalled();
    });
});
