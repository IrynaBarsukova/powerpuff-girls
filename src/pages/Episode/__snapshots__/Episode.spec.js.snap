// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episode /> should render Episode 1`] = `
<div>
  <div
    class="episode"
  >
    <div
      class="container"
    >
      <button
        class="episode__button"
        type="button"
      >
        Go back
      </button>
      <div
        class="entity"
      >
        <h2
          class="entity__title"
        >
          Bye Bye, Bellum
        </h2>
        <section
          class="entity__details"
        >
          <img
            alt="Bye Bye, Bellum"
            class="entity__image"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/53/132631.jpg"
          />
          <p
            class="entity__summary"
          >
            <p>
              The Mayor of Townsville overworks the Powerpuff Girls after his secretary Sara Bellum quits
            </p>
          </p>
        </section>
      </div>
    </div>
  </div>
</div>
`;
